lib/utils/dev-to.ts

```diff
@@ -0,0 +1,13 @@
+export const getBlogDetails = async (blogLink: string) => {
+  const trimmedUrl = blogLink.trim();
+  const devToUrl = new URL(trimmedUrl.includes("https://") ? trimmedUrl : `https://${trimmedUrl}`);
+  const { pathname } = devToUrl;
+  const [, username, slug] = pathname.split("/");
+  const response = await fetch(`https://dev.to/api/articles/${username}/${slug}`);
```

Comment:
What is happening here? Can you walk through why the above us need and we can't fetch the url provided?

Reply:
This is the same as [this](https://github.com/diivi/insights/blob/e1b524572f907d4ad2a903fee8a6c145f6c50588/lib/utils/github.ts#L28)